src/nativefuncs.c

```diff
@@ -0,0 +1,125 @@
+/**
+ * @file nativefuncs.c
+ * @author Derek Tan
+ * @brief Implements Rubel language's native functions within modules "io", "lists", etc. These are bound on startup to interpreter function "groups" which represent native or user-defined modules.
+ * @date 2023-08-01
+ */
+
+#include "backend/api/natives/nativefuncs.h"
+
+/// SECTION: module io
+
+VarValue *rubel_print(FuncArgs *args)
+{
+    VarValue *arg1 = funcargs_get_at(args, 0);
+    DataType arg1_type = arg1->type;
+
+    if (!arg1) return NULL;
+
+    switch (arg1_type)
+    {
+    case INT_TYPE:
+        printf("%i", arg1->data.int_val.value);
+        break;
+    case REAL_TYPE:
+        printf("%f", arg1->data.real_val.value);
+        break;
+    case STR_TYPE:
+        printf("%s", arg1->data.str_type.value->source);
+        break;
+    case BOOL_TYPE:
+        if (arg1->data.bool_val.flag) printf("boolean($T)");
+        else printf("boolean($F)");
+        break;
+    case LIST_TYPE:
+        printf("list[%zu]", arg1->data.list_type.value->count);
+        break;
+    default:
+        break;
+    }
+
+    return NULL; // return the argument if it existed anyways... tells interpreter that print was OK
+}
+
+VarValue *rubel_println(FuncArgs *args)
+{
+    VarValue *arg1 = funcargs_get_at(args, 0);
+    DataType arg_type = varval_get_type(arg1);
+
+    if (!arg1) return NULL;
+
+    switch (arg_type)
+    {
+    case INT_TYPE:
+        printf("%i\n", arg1->data.int_val.value);
+        break;
+    case REAL_TYPE:
+        printf("%f\n", arg1->data.real_val.value);
+        break;
+    case STR_TYPE:
+        printf("%s\n", arg1->data.str_type.value->source);
+        break;
+    case BOOL_TYPE:
+        if (arg1->data.bool_val.flag) printf("boolean($T)\n");
+        else printf("boolean($F)\n");
+        break;
+    case LIST_TYPE:
+        printf("list[%zu]\n", arg1->data.list_type.value->count);
+        break;
+    default:
+        break;
+    }
+
+    return NULL;
+}
+
+VarValue *rubel_input(FuncArgs *args)
```

Comment:
This function is still untested. I must add a new to-do to remind me to create a `"cat"` program test for this native function. Such a program will repeat the user inputted string given in `stdin`.